### ISSUE TITLE
"Active Note" fails with "API key not found"

### ISSUE THREAD

**Issue by Bouke** [logancyang/obsidian-copilot#140]
I'm trying to chat about my 'active note'. I've setup all the Azure OpenAI properties. Normal chat works, however when I select 'QA: Active Note', the following error appears in the developer console:

```
plugin:copilot:60197 Uncaught (in promise) Error: OpenAI or Azure OpenAI API key not found
    at new OpenAIEmbeddings (plugin:copilot:60197:13)
    at AIState.getEmbeddingsAPI (plugin:copilot:60657:33)
    at AIState.buildIndex (plugin:copilot:60842:32)
    at async AIState.setChain (plugin:copilot:60811:11)
```

When I ask about 'this note', I get a non-helpful response:

![image](https://github.com/logancyang/obsidian-copilot/assets/235882/a80ea428-3b23-4c4d-a24a-abd30fd57f94)

**Comment by logancyang** [comment]
My guess is that your embedding provider is still OpenAI and not Azure, and you don't have OpenAI key in your settings. Could you pls check?

**Comment by kanawish** [comment]
fwiw, I have azure fully configured in my settings, and I'm getting this message when rebuilding the index. (And only when rebuilding)
I added breakpoints, and it seems that the `azureApiKey` is not being passed along in the code.

**Comment by kanawish** [comment]
It might just be a misunderstanding on the UX, it looks like this error only takes place when 'conversation' is selected in the dropdown box next to the 'rebuild' button.

**Comment by logancyang** [comment]
Released v2.4.14, let me know if this is still an issue. Closing for now
